src/SourceBuild/content/eng/tools/CreateBaselineUpdatePR/PRCreator.cs

```diff
@@ -280,72 +280,120 @@ private async Task<TreeResponse> CreateParentTreeAsync(TreeResponse testResultsT
         {
-            var comparison = await _client.Repository.Commit.Compare(_repoOwner, _repoName, headReference.Object.Sha, commitResponse.Sha);
-            if (!comparison.Files.Any())
+            string pullRequestBody = $"This PR was created by the `CreateBaselineUpdatePR` tool for build {buildId}. \n\n" +
+                                 $"The updated test results can be found at {BuildLink}{buildId} (internal Microsoft link)";
+            if (existingPullRequest != null)
             {
-                // No changes to commit
-                Log.LogInformation("No changes to commit. Skipping PR creation.");
-                return;
+                await UpdatePullRequestAsync(newBranchName, commitSha, pullRequestBody, existingPullRequest);
             }
+            else
+            {
+                await CreatePullRequestAsync(newBranchName, commitSha, targetBranch, title, pullRequestBody);
+            }
+        }
+    }
 
-            // Create a new pull request
-            var newReference = new NewReference("refs/heads/" + newBranchName, commitResponse.Sha);
-            await _client.Git.Reference.Create(_repoOwner, _repoName, newReference);
+    private async Task<PullRequest?> GetExistingPullRequestAsync(string title, string targetBranch)
+    {
+        var request = new PullRequestRequest
+        {
+            Base = targetBranch
+        };
+        var existingPullRequest = await _client.PullRequest.GetAllForRepository(_repoOwner, _repoName, request);
+        return existingPullRequest.FirstOrDefault(pr => pr.Title == title);
+    }
 
-            var newPullRequest = new NewPullRequest(title, newBranchName, targetBranch)
-            {
-                Body = pullRequestBody
-            };
-            var pullRequest = await _client.PullRequest.Create(_repoOwner, _repoName, newPullRequest);
+    private async Task<string> CreateCommitAsync(string newSha, string headSha, string commitMessage)
+    {
+        var newCommit = new NewCommit(commitMessage, newSha, headSha);
+        var commit = await _client.Git.Commit.Create(_repoOwner, _repoName, newCommit);
+        return commit.Sha;
+    }
 
-            Log.LogInformation($"Created pull request #{pullRequest.Number}. URL: {pullRequest.HtmlUrl}");
+    private async Task<bool> ShouldMakeUpdatesAsync(string headSha, string commitSha)
+    {
+        var comparison = await _client.Repository.Commit.Compare(_repoOwner, _repoName, headSha, commitSha);
+        if (!comparison.Files.Any())
+        {
+            Log.LogInformation("No changes to commit. Skipping PR creation/updates.");
+            return false;
         }
+        return true;
+    }
+
+    private async Task UpdatePullRequestAsync(string branchName, string commitSha, string body, PullRequest pullRequest)
+    {
+        await UpdateReferenceAsync(branchName, commitSha);
+
+        var pullRequestUpdate = new PullRequestUpdate
+        {
+            Body = body
+        };
```

Comment:
What does this actually do in the context of an existing PR. Does it add a comment? Is that necessary? It looks like the same content as the original PR message.

Reply:
The body needs to be updated because it includes a link to the corresponding run of the pipeline.

Reply:
I see. So it replaces the content of the original PR body?

Reply:
Yep! An example is https://github.com/dotnet/sdk/pull/41514 where the tool updated the body:

![image](https://github.com/dotnet/sdk/assets/67609881/9c87441a-ae04-45ca-a8f5-80e96435f6d5)